sky/skylet/providers/gcp/config.py

```diff
@@ -506,10 +507,50 @@ def _configure_subnet(config, compute):
     ):
         return config
 
-    subnets = _list_subnets(config, compute)
+    subnets = _list_subnets(config, compute, filter=f'(name="{SKYPILOT_VPC_NAME}")')
 
     if not subnets:
-        raise NotImplementedError("Should be able to create subnet.")
+        # Create a default VPC network and firewall rules
+        proj_id = config["provider"]["project_id"]
+
+        logger.info(f'Creating a default VPC network, {SKYPILOT_VPC_NAME}...')
+        vpc_network_body = {
+            "name": SKYPILOT_VPC_NAME,
+            "selfLink": f"projects/{proj_id}/global/networks/{SKYPILOT_VPC_NAME}",
+            "autoCreateSubnetworks": True,
+            "mtu": 1460,
+            "routingConfig": {"routingMode": "GLOBAL"},
```

Comment:
For future maintenance, can we add some comments for these arguments, such as where we get these argument value?

Reply:
Good point! These are default parameters from GCP. Added comments to `constants.py`
```
# Below parameters are from the default VPC on GCP.
# https://cloud.google.com/vpc/docs/firewalls#more_rules_default_vpc
```